src/core/solvers/ddp.cpp

```diff
@@ -438,28 +447,61 @@ const std::vector<Eigen::VectorXd>& SolverDDP::get_k() const { return k_; }
 
 const std::vector<Eigen::VectorXd>& SolverDDP::get_fs() const { return fs_; }
 
-void SolverDDP::set_regfactor(const double& regfactor) {
+void SolverDDP::set_reg_incfactor(double regfactor) {
+  if (regfactor <= 1.) {
+    throw_pretty("Invalid argument: "
+                 << "reg_incfactor value is higher than 1.");
+  }
+  reg_incfactor_ = regfactor;
+}
+
+void SolverDDP::set_reg_decfactor(double regfactor) {
+  if (regfactor <= 1.) {
+    throw_pretty("Invalid argument: "
+                 << "reg_decfactor value is higher than 1.");
+  }
+  reg_decfactor_ = regfactor;
+}
+
+void SolverDDP::set_regfactor(double regfactor) {
   if (regfactor <= 1.) {
     throw_pretty("Invalid argument: "
                  << "regfactor value is higher than 1.");
   }
-  regfactor_ = regfactor;
+  reg_incfactor_ = regfactor;
+  reg_decfactor_ = regfactor;
+}
+
+void SolverDDP::set_reg_min(double regmin) {
+  if (0. > regmin) {
+    throw_pretty("Invalid argument: "
+                 << "regmin value has to be positive.");
+  }
+  reg_min_ = regmin;
 }
 
-void SolverDDP::set_regmin(const double& regmin) {
+void SolverDDP::set_regmin(double regmin) {
   if (0. > regmin) {
```

Comment:
should be changed following https://github.com/loco-3d/crocoddyl/pull/893#discussion_r542122288